shotgun_api3/shotgun.py

```diff
@@ -675,6 +688,32 @@ def _construct_read_parameters(self,
         params["paging"] = { "entities_per_page": self.config.records_per_page,
                              "current_page": 1 }
 
+        if additional_server_presets is not None:
+            if not isinstance(additional_server_presets, list):
+                msg = "'additional_server_presets' parameter must be a list or None"
+                raise ValueError(msg)
+
+            preset_list = []
+            for preset in additional_server_presets:
+                if not isinstance(preset, list):
+                    msg = "Elements of 'additional_server_presets' parameter must be lists"
+                    raise ValueError(msg)
+
+                if len(preset) == 0:
+                    msg = "Missing 'name' for 'additional_server_presets'"
+                    raise ValueError(msg)
+
+                preset_param = {
+                    'name': preset[0]
+                }
+
+                if len(preset) > 1:
+                    preset_param['params'] = preset[1]
```

Comment:
I was thinking of something like this:
https://github.com/shotgunsoftware/python-api/blob/35229_add_additional_server_presets/shotgun_api3/shotgun.py#L2255